src/redux/actions/genes/loadGeneExpression.js

```diff
@@ -36,17 +36,20 @@ const loadGeneExpression = (
 
   let genesToFetch = [...genes];
   const genesAlreadyLoaded = Object.keys(geneData);
-
   if (!forceReloadAll) {
     genesToFetch = genesToFetch.filter(
       (gene) => !new Set(upperCaseArray(genesAlreadyLoaded)).has(gene.toUpperCase()),
     );
   }
 
-  const upperGenesAlreadyLoaded = new Set(upperCaseArray(genesAlreadyLoaded));
-  const displayedGenes = genes.filter((gene) => upperGenesAlreadyLoaded.has(gene.toUpperCase()));
-
+  const displayedGenes = genesAlreadyLoaded.filter(
+    (gene) => upperCaseArray(genes).includes(gene.toUpperCase()),
+  );
   if (genesToFetch.length === 0) {
+    // All genes are already loaded.
+    if (onUpdate) {
+      onUpdate({ shownGene: displayedGenes[0] });
+    }
```

Comment:
Please move this out of this action creator.  This creates a tight coupling between action creator and updating the plot. You can return the displayed gene from the action and use trigger the update outside of this action creator.